treeherder/webapp/api/performance_data.py

```diff
@@ -693,26 +693,30 @@ def list(self, request):
         serialized_data = serializer.data
 
         if no_retriggers:
-            serialized_data = self.__filter_out_retriggers(serialized_data)
+            serialized_data = self._filter_out_retriggers(serialized_data)
 
         return Response(data=serialized_data)
 
     @staticmethod
-    def __filter_out_retriggers(serialized_data: List[dict]) -> List[dict]:
+    def _filter_out_retriggers(serialized_data: List[dict]) -> List[dict]:
         """
         Removes data points resulted from retriggers
         """
+
         for perf_summary in serialized_data:
-            deletable_idxs, seen_push_id = [], None
+            retrigger_idxs, seen_push_ids = [], []
             for idx, datum in enumerate(perf_summary['data']):
-                if seen_push_id == datum['push_id']:
-                    deletable_idxs.append(idx)
+                if datum['push_id'] in seen_push_ids:
+                    retrigger_idxs.append(idx)
                 else:
-                    seen_push_id = datum['push_id']
-
-            if deletable_idxs:
-                start, end = deletable_idxs[0], deletable_idxs[-1]
-                del perf_summary['data'][start : end + 1]
+                    seen_push_ids.append(datum['push_id'])
```

Comment:
I'm not sure why `seen_push_id` wasn't enough to figure out retriggered data points.